### PR TITLE
Add SPI imports for SwiftCorelibsFoundation usage of Decimal

### PR DESCRIPTION
**Pull request by parkera**
Also adds some tests for the exported functionality that SCL-F uses.

### CODE REVIEW

**Comment by parkera** [comment]
@swift-ci test

**Review by iCharlesHu — APPROVED** [review]

File: Sources/FoundationEssentials/Decimal/Decimal+Conformances.swift [hunk 1]
```diff
@@ -14,6 +14,130 @@
 internal import _ForSwiftFoundation
 #endif // FOUNDATION_FRAMEWORK
 
+/*
```
Comment (iCharlesHu) [inline Sources/FoundationEssentials/Decimal/Decimal+Conformances.swift:17]: Are we leaving these as comments intentionally?

Comment (parkera) [reply Sources/FoundationEssentials/Decimal/Decimal+Conformances.swift:17]: Nope, thanks. Will fix.

**Comment by parkera** [comment]
@swift-ci test

**Pull request merged**